### PR TITLE
Use standard role for custom build

### PR DESCRIPTION
**Pull request by vrutkovs**
Turns out there was no point creating an additional role, as there 
has always been a standard one there. This change would work on 
both old (3.4 - 3.6) and new (3.7+) clusters

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -32,46 +32,6 @@
...
-  command: oc get policybinding {{ osbs_namespace }}:default --namespace {{ osbs_namespace }}
-  environment: "{{ osbs_environment }}"
-  register: policybinding_query
-  failed_when: policybinding_query.rc != 0 and ('not found' not in policybinding_query.stderr)
-  changed_when: false
-  when: osbs_is_admin
-  tags:
-  - oc
-
-- name: create a policybinding
```
Comment (lcarva) [inline tasks/main.yml]: Isn't this needed initially? I've had problems creating roles/rolebindings if this is not called first.

File: tasks/main.yml [hunk 2]
```diff
@@ -98,16 +58,12 @@
     groups: "{{ osbs_admin_groups }}"
 
   - name: osbs-custom-build-readwrite
-    role: osbs-custom-build
-    yaml_version: v1
-    role_namespace: "{{ osbs_namespace }}"
+    role: system:build-strategy-custom
```
Comment (lcarva) [inline tasks/main.yml]: Doesn't using clusterroles require elevated privileges?

File: tasks/main.yml (hunk 1 above)
Comment (lcarva) [reply tasks/main.yml]: It might be ok because we're not binding to project roles, but instead cluster roles.

Comment (vrutkovs) [reply tasks/main.yml]: We are binding this to a project namespace, but a new policybinding is not required for that - as a new custom role is not created anymore

File: tasks/main.yml (hunk 2 above)
Comment (vrutkovs) [reply tasks/main.yml]: The role is defined in a cluster, yes, but the binding happens per namespace (as `RoleBinding` is default, unlike `pruner) - admin would be able to run custom builds on current namespace only, not the whole cluster.

I'm not sure if that would work on OSD right away, but we could ask for a permission to run custom builds, as the changes would be limited to the namespace only.

**Comment by rcerven** [comment]
+1

**Comment by lcarva** [comment]
@twaugh, @rcerven ok to merge this? It's required for 3.7 (which is the version of the new cluster we're deploying)

**Comment by lcarva** [comment]
This actually causes a problem with 3.6. We use the format `system:serviceaccount:my-namespace:spam` to refer to a service account in `osbs_readwrite_users`, for instance. With this change, it causes the yaml to be generated in the format:

```
- kind: 'User'
  namespace: worker-test
  name: system:serviceaccount:my-namespace:spam
```

This is invalid:
```
Invalid value: "system:serviceaccount:my-namespace:spam": may not contain ":\"
```

It also causes issues if service account name contains `"/"`

**Comment by lcarva** [comment]
See https://github.com/projectatomic/ansible-role-osbs-namespace/pull/25 for an additional commit that works on both 3.6 and 3.7. (tested).

**Comment by twaugh** [comment]
Superseded by #25.

**Pull request closed without merging**